grudge/models/wave.py

```diff
@@ -221,11 +221,11 @@ def __init__(self, dcoll, c, source_f=None,
 
         actx = c.array_context
         self.dcoll = dcoll
-        self.c = c
+        self.c = actx.freeze(c)
```

Comment:
Since `c` is documented to come in frozen, we should probably get rid of the access to `c.array_context`, in keeping with https://github.com/inducer/arraycontext/issues/162.

Reply:
Yep, done! Also asserted that `c` is actually received as a frozen DOFArray.